app/src/main/java/mozilla/lockbox/presenter/AutofillRoutePresenter.kt

```diff
@@ -91,51 +91,11 @@ class AutofillRoutePresenter(
         }
     }
 
-    private fun navigateToFragment(@IdRes destinationId: Int, args: Bundle? = null) {
-        val src = navController.currentDestination ?: return
-        val srcId = src.id
-        if (srcId == destinationId && args == null) {
-            // No point in navigating if nothing has changed.
-            return
-        }
-
-        val transition = findTransitionId(srcId, destinationId) ?: destinationId
-
-        if (transition == destinationId) {
-            // Without being able to detect if we're in developer mode,
-            // it is too dangerous to RuntimeException.
-            val from = activity.resources.getResourceName(srcId)
-            val to = activity.resources.getResourceName(destinationId)
-            log.error(
-                "Cannot route from $from to $to. " +
-                    "This is a developer bug, fixable by adding an action to graph_autofill.xml"
-            )
-        } else {
-            val clearBackStack = src.getAction(transition)?.navOptions?.shouldLaunchSingleTop() ?: false
-            if (clearBackStack) {
-                while (navController.popBackStack()) {
-                    // NOP
-                }
-            }
-        }
-
-        try {
-            navController.navigate(transition, args)
-        } catch (e: IllegalArgumentException) {
-            log.error("This appears to be a bug in navController", e)
-            navController.navigate(destinationId, args)
-        }
-    }
-
-    private fun findTransitionId(@IdRes from: Int, @IdRes to: Int): Int? {
-        return when (Pair(from, to)) {
-            Pair(R.id.fragment_locked, R.id.fragment_filter) -> R.id.action_locked_to_filter
-            Pair(R.id.fragment_null, R.id.fragment_filter) -> R.id.action_to_filter
-            Pair(R.id.fragment_null, R.id.fragment_locked) -> R.id.action_to_locked
-            else -> null
-        }
+    private fun navigateTo(@IdRes destinationId: Int, args: Bundle? = null) {
+        super.navigateToFragment(navController, destinationId, args)
```

Comment:
is there a better way to pass the `navController`?

Reply:
yes, let's have it be a `public` property of the superclass so that we don't have to hold onto this `navigateTo` method that just calls `super`.